lib/identity/helpers/auth.rb

```diff
@@ -168,7 +168,76 @@ def perform_oauth_refresh_dance
 
         log :oauth_refresh_dance_complete, session_id: @cookie.session_id
       end
+    end
+
+    # Attempt to resolve an oauth/authorize request
+    def call_authorize(authorize_params=get_authorize_params)
+      # clear anything that might be left over in the session
+      @cookie.authorize_params = nil
 
+      begin
+        # Try to perform an access token refresh if we know it's expired. At
+        # the time of this writing, refresh tokens last 30 days (much longer
+        # than the short-lived 2 hour access tokens).
+        if Time.now > @cookie.access_token_expires_at
```

Comment:
Do not use `Time.now` without zone. Use one of `Time.zone.now`, `Time.now.current`, `Time.now.in_time_zone`, `Time.now.utc`, `Time.now.getlocal`, `Time.now.iso8601`, `Time.now.jisx0301`, `Time.now.rfc3339`, `Time.now.to_i`, `Time.now.to_f` instead.